### ISSUE TITLE
Fix FP S3699 (`no-use-of-empty-return-value`): Ignore ambient functions

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#4579]
The rule is reporting false positives on TypeScript ambient functions:

```typescript
declare function foo(bar: number): number;
let x = foo(42); // False positive
```

https://community.sonarsource.com/t/typescript-false-positive-for-typescript-s3699-when-using-declare-function/109601
